core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -266,8 +266,21 @@ private[spark] class TaskSchedulerImpl(
               assert(availableCpus(i) >= 0)
               launchedTask = true
             }
+          } catch {
+            case e: TaskNotSerializableException => {
+              logError(s"Resource offer failed, task set ${taskSet.name} was not serializable")
+              // Do not offer resources for this task, but don't throw an error to allow other
+              // task sets to be submitted.
+              return
+            }
           }
         }
+      }
+    }
```

Comment:
can you define this function as a `private def` outside of `resourceOffers`? The nesting here makes this hard to read.